doc/fwutil/fwutil.md

```diff
@@ -462,6 +519,321 @@ Aborted!
+    - response : installed firmware version
+2. {utility} -n(--notification) : able to provide the update complete action :
+    - equivalent to the component api's get_firmware_update_notification(image_path) 
+    - response : the required action to complete the component installation
+3. {utility} -a(--autoupdate) : able to perform the auto-update action :
+    - equivalent to the component api's auto_update_firmware(image_path, boot_type) 
+    - auto-update interface needs two arguments : image_path and boot_type
+    - response : the result of auto-update as follows
+        ```json
+        "True : updated"
+        "True : scheduled"
+        "True : installed"
+        "False : ${ErrorCode}"
+        ```
+
+**Optional) The utility can be supported for other platform api substitues like `compoenent_update` and `compoenent_install` with |-u(--update)|-i(--install)**
+
+The component utility needs to be called by the FWutil command to perform the firmware auto-update process if it's defined in the `platform_components.json`, otherwise, the platform component api will be called.
+The componet utility path will be pased from the `platform_components.json` and be executed by fwutil.
+Below shows how the utility can be executed for the auto-update interface.
+```bash
+...
+firmware_path = parser[self.__pcp.FIRMWARE_KEY]
+utility_path = parser[self.__pcp.UTILITY_KEY]
+if os.path.isfile(utility_path) and os.access(utility_path, os.X_OK):
+    cmd = "{} -a {} {}".format(
+        utility_path,
+        firmware_path,
+        boot_type
+    )
+    output = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
+...
+```
+From the above `platform_components.json` example, if the platform needs the SSD firmware update, then the auto-update can be triggered with following command.
+`$PWD/ssd_fw_update -a $PWD/SSD.bin fast` 
+
+##### 2.2.2.4.3.2 Platform Firmware Update Reboot Handle Plugin
+The reboot scripts will invoke the platform firmware update reboot plugin with boot-type, which will analyze the reboot firmware task file and execute the upgrade commands for the components present in task file.
+```bash
+PLATFORM_FWUTIL_AU_REBOOT_HANDLE="platform_fwutil_au_reboot_handle"
+if [[ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_FWUTIL_AU_REBOOT_HANDLE} ]]; then
+    debug "Handling task file for boot type ${REBOOT_TYPE}”
+    ${DEVPATH}/${PLATFORM}/${PLATFORM_FWUTIL_AU_REBOOT_HANDLE} ${REBOOT_TYPE}
+fi
+```
+
+##### 2.2.2.4.4 Auto-update Use Cases
+
+##### 2.2.2.4.4.1 Standalone firmware update during run time
+Since FWutil auto-update can support the firmware update using a custom firmware package, this will allow the firmware update to be available during run time.
+
+SONiC device can have a component firmware updated without interrupting the data plan if the component firmware update doesn't need any boot action.
+```bash
+root@sonic:~# fwutil auto-update fw --fw-image=aboot-fw-update.tar.gz --boot=none
```

Comment:
@sujinmkang What if the `cold/fast/warm` boot was specified as the desired action but some of the platform components do support firmware upgrade without dataplane interruption? Do we expect the update will take place implicitly or we still do want to do only with `--boot=none`?

Reply:
That is good question. Without boot option, the update which can be done without data plane interruption, should be performed implicitly. I will add the default setting for boot type.